dds/DCPS/security/AccessControlBuiltInImpl.cpp

```diff
@@ -1658,92 +1652,68 @@ namespace {
   const TimeDuration MAX_DURATION = TimeDuration(3600, 0);
 }
 
-bool
-AccessControlBuiltInImpl::RevokePermissionsTimer::start_timer(
-  const time_t& expiration, ::DDS::Security::PermissionsHandle pm_handle)
+void
+AccessControlBuiltInImpl::RevokePermissionsTask::insert(::DDS::Security::PermissionsHandle pm_handle,
+                                                        const time_t& expiration)
 {
-  ACE_GUARD_RETURN(ACE_Thread_Mutex, guard, lock_, false);
+  ACE_GUARD(ACE_Thread_Mutex, guard, lock_);
 
   const time_t current_date_time = time(0);
   tm* current_time_tm = gmtime(&current_date_time);
   const time_t cur_utc_time = mktime(current_time_tm);
   const TimeDuration duration = std::min(TimeDuration(expiration - cur_utc_time), MAX_DURATION);
 
-  Entry* entry = new Entry(pm_handle, expiration);
-
-  ACE_Reactor_Timer_Interface* reactor = TheServiceParticipant->timer();
+  const bool empty_before = expiration_to_handle_.empty();
 
-  if (reactor) {
-    const long timer_id = reactor->schedule_timer(this, entry, duration.value());
-
-    if (timer_id != -1) {
-      entry_map_[pm_handle] = entry;
-      return true;
-    }
+  ExpirationToHandle::const_iterator pos =
+    expiration_to_handle_.insert(ExpirationToHandle::value_type(expiration, pm_handle));
 
+  if (!empty_before && pos == expiration_to_handle_.begin()) {
+    cancel();
   }
 
-  delete entry;
-  return false;
+  if (pos == expiration_to_handle_.begin()) {
+    schedule(duration);
+  }
 }
 
-int
-AccessControlBuiltInImpl::RevokePermissionsTimer::handle_timeout(
-  const ACE_Time_Value& /*tv*/, const void* arg)
+void
+AccessControlBuiltInImpl::RevokePermissionsTask::execute(const DCPS::MonotonicTimePoint& /*now*/)
```

Comment:
Speaking of TimePoint... is this the right time to move some of this time_t calc stuff over to use our TimePoint classes? @iguessthislldo any opinions on this lingering time_t usage?